controllers/imagecollector/imagecollector_controller.go

```diff
@@ -222,11 +274,146 @@ func (r *Reconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Resu
 	return ctrl.Result{RequeueAfter: *repeatPeriod}, nil
 }
 
-func isNotFound(err error) bool {
-	if err != nil && client.IgnoreNotFound(err) == nil {
-		return true
+func (r *Reconciler) getChildImageJobs(ctx context.Context, collector *eraserv1alpha1.ImageCollector) ([]eraserv1alpha1.ImageJob, error) {
+	imageJobList := &eraserv1alpha1.ImageJobList{}
+	if err := r.List(ctx, imageJobList); err != nil {
+		log.Info("could not list imagejobs")
+		return nil, err
+	}
+
+	relevantJobs := util.FilterJobListByOwner(
+		imageJobList.Items, metav1.NewControllerRef(collector, schema.GroupVersionKind{
+			Group:   "eraser.sh",
+			Version: "v1alpha1",
+			Kind:    "ImageCollector",
+		}),
+	)
+
+	return relevantJobs, nil
+}
+
+func (r *Reconciler) upsertImageList(ctx context.Context, collector *eraserv1alpha1.ImageCollector) (ctrl.Result, error) {
+	imageListItems := make([]string, 0, len(collector.Status.Vulnerable))
+	images := collector.Status.Vulnerable
+
+	if *deleteScanFailedImages {
+		images = append(images, collector.Status.Failed...)
+	}
+
+	for i := range images {
+		img := images[i]
+		imageListItems = append(imageListItems, img.Digest)
+	}
+
+	imageList := eraserv1alpha1.ImageList{}
+
+	err := r.Get(ctx, types.NamespacedName{Namespace: "", Name: "imagelist"}, &imageList)
+	if isNotFound(err) {
+		return r.createImageList(ctx, collector, imageListItems)
+	}
+
+	// else update
+	imageList.Spec.Images = imageListItems
```

Comment:
In this case it seems like we are clobbering whatever imagelist is already there.
Is this how we are wanting this to work?

Reply:
@cpuguy83 We discussed this in the scrum and yes, that's intentional. We also modified the behavior of the collector to only supply nonrunning images. The idea is that the collector gets a snapshot of nonrunning images, that snapshot is filtered down by the scanner to produce the final imagelist. We don't support multiple imagelists so we clobber. Without clobbering, the imagelist might represent the state of the cluster yesterday plus the state of the cluster today (if we're on a 24-hour timer).

The only situation I can imagine where this would be a problem is if someone has created an imagelist manually just before the timed collect->scan->erase pipeline begins and they conflict with each other. But that should be solved by #180